src/main/java/duke/util/Parser.java

```diff
@@ -0,0 +1,123 @@
+package duke.util;
+
+import duke.exception.DukeException;
+import duke.exception.InvalidCommandException;
+import duke.exception.NoDateProvidedException;
+import duke.exception.NoDescriptionException;
+
+import java.io.IOException;
+
+/**
+ * Parser
+ *
+ * <p>CS2103T AY19/20 Semester 2
+ * Individual project
+ * Duke project
+ *
+ * 30 Jan 2020
+ *
+ * @author Jel
+ */
+public class Parser {
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a Parser instance.
+     * @param tasks List where tasks are to be stored.
+     * @param ui User interface of the program.
+     */
+    public Parser(TaskList tasks, Ui ui) {
+        this.tasks = tasks;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses user input.
+     * @param line The line of input to be parsed.
+     */
+    public void parseInput(String line) {
+        String trimmed = line.trim();
+        String cmd = getCommand(trimmed);
+        String descAndDate;
+
+        try {
+            switch (cmd) {
+            case "bye":
+                ui.bye();
+                return;
+            case "todo":
+                descAndDate = getDescAndDate(cmd, trimmed);
+                tasks.addTodo(descAndDate);
+                break;
+            case "deadline":
+                descAndDate = getDescAndDate(cmd, trimmed);
+                tasks.addDeadlineOrEvt(" /by ", descAndDate);
+                break;
+            case "event":
+                descAndDate = getDescAndDate(cmd, trimmed);
+                tasks.addDeadlineOrEvt(" /at ", descAndDate);
+                break;
+            case "list":
+                tasks.listTasks();
+                break;
+            case "done":
+                tasks.markTaskAsDone(trimmed);
+                break;
+            case "delete":
+                tasks.deleteTask(trimmed);
+                break;
+            case "find":
+                tasks.findTask(trimmed);
+                break;
+            default:
+                throw new InvalidCommandException();
```

Comment:
Good indentation but there is lack of abstraction for your different commands here.

Should your Parser class be handling the logic of executing the commands? It should only be parsing the user inputs.

Think of how you can abstract the different commands (delete, todo, event, ...) into separate classes and let them handle their own command execution respectively.